pkg/reconciler/revision/resources/deploy.go

```diff
@@ -137,53 +137,45 @@ func rewriteUserProbe(p *corev1.Probe, userPort int) {
 
 func makePodSpec(rev *v1alpha1.Revision, loggingConfig *logging.Config, tracingConfig *tracingconfig.Config, observabilityConfig *metrics.ObservabilityConfig, autoscalerConfig *autoscaler.Config, deploymentConfig *deployment.Config) (*corev1.PodSpec, error) {
 	queueContainer, err := makeQueueContainer(rev, loggingConfig, tracingConfig, observabilityConfig, autoscalerConfig, deploymentConfig)
-
 	if err != nil {
 		return nil, fmt.Errorf("failed to create queue-proxy container: %w", err)
 	}
 
-	userContainer := rev.Spec.GetContainer().DeepCopy()
-	// Adding or removing an overwritten corev1.Container field here? Don't forget to
-	// update the fieldmasks / validations in pkg/apis/serving
-
-	userContainer.VolumeMounts = append(userContainer.VolumeMounts, varLogVolumeMount)
-	userContainer.Lifecycle = userLifecycle
-	userPort := getUserPort(rev)
-	userPortInt := int(userPort)
-	userPortStr := strconv.Itoa(userPortInt)
-	// Replacement is safe as only up to a single port is allowed on the Revision
-	userContainer.Ports = buildContainerPorts(userPort)
-	userContainer.Env = append(userContainer.Env, buildUserPortEnv(userPortStr))
-	userContainer.Env = append(userContainer.Env, getKnativeEnvVar(rev)...)
-	// Explicitly disable stdin and tty allocation
-	userContainer.Stdin = false
-	userContainer.TTY = false
-
-	// Prefer imageDigest from revision if available
-	if rev.Status.ImageDigest != "" {
-		userContainer.Image = rev.Status.ImageDigest
+	containers := []corev1.Container{
+		*queueContainer,
 	}
 
-	if userContainer.TerminationMessagePolicy == "" {
-		userContainer.TerminationMessagePolicy = corev1.TerminationMessageFallbackToLogsOnError
+	// closure function for serving container
+	servingContainer := func() {
```

Comment:
This closure feels a little odd. Maybe create add the resolution code to `makeServingContainer` and create a new `makeSidecarContainer` which does the resolution for sidecars.

The append call can be "duplicated". It'll be needed only twice anyway.